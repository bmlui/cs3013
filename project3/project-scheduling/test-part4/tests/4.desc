analysis: RR: Properly handle idle CPUs for multiple times

