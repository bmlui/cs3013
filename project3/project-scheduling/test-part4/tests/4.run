./scheduler 1 RR tests/4.in 2
