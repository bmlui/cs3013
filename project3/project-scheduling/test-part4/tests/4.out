Execution trace with RR:
t=0: [Job 0] arrived at [0], ran for: [1]
t=2: [Job 1] arrived at [2], ran for: [2]
t=4: [Job 2] arrived at [3], ran for: [2]
t=6: [Job 3] arrived at [5], ran for: [2]
t=8: [Job 4] arrived at [7], ran for: [1]
t=9: [Job 1] arrived at [2], ran for: [2]
t=11: [Job 2] arrived at [3], ran for: [2]
t=13: [Job 3] arrived at [5], ran for: [2]
t=15: [Job 1] arrived at [2], ran for: [1]
t=16: [Job 2] arrived at [3], ran for: [2]
t=18: [Job 2] arrived at [3], ran for: [2]
t=20: [Job 2] arrived at [3], ran for: [2]
t=22: [Job 2] arrived at [3], ran for: [1]
t=25: [Job 5] arrived at [25], ran for: [2]
t=27: [Job 5] arrived at [25], ran for: [2]
End of execution with RR.
Begin analyzing RR:
Job 0 -- Response time: 0  Turnaround: 1  Wait: 0
Job 1 -- Response time: 0  Turnaround: 14  Wait: 9
Job 2 -- Response time: 1  Turnaround: 20  Wait: 9
Job 3 -- Response time: 1  Turnaround: 10  Wait: 6
Job 4 -- Response time: 1  Turnaround: 2  Wait: 1
Job 5 -- Response time: 0  Turnaround: 4  Wait: 0
Average -- Response: 0.50  Turnaround 8.50  Wait 4.17
End analyzing RR.
